with-go/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"context"
+	"html/template"
+	"log"
+	"net/http"
+
+	"github.com/machinebox/graphql"
+)
+
+func main() {
+	http.HandleFunc("/", getProducts)
+	http.ListenAndServe(":8080", nil)
```

Comment:
This function returns an error if the server could not be started up, you could check it with an if statement and conditionally log or panic to make sure the user knows when something's going wrong!

```go
err := http.ListenAndServe(":8080", nil)
if err != nil {
  panic(err)
}
```